src/lib/utils/flattenAndCategorizeErrors.js

```
@@ -0,0 +1,51 @@
+// This file is part of React-Invenio-Forms
+// Copyright (C) 2025 CERN.
+//
+// React-Invenio-Forms is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+export function flattenAndCategorizeErrors(obj, prefix = "") {
+  let flattenedErrors = {}; // Store primitive errors
+  let severityChecks = {}; // Store errors with severity (e.g., warnings, errors)
+
+  // Handle cases where the input object itself contains message and severity
+  if ("message" in obj && "severity" in obj) {
```

if those keys are not in the object, we should raise an error, not hide it by conditional statements. It means we are getting unexpected structure of the error, and without clear fail, it will be hard to understand what is happening while debugging